edisgo/grid/components.py

```diff
@@ -204,11 +207,49 @@ def timeseries(self):
                         "No timeseries for load of type {} "
                         "given.".format(sector))
                     raise
-            ts['q'] = ts['p'] * q_factor
-            return ts * consumption
+            ts * consumption
+            if self.timeseries_reactive is not None:
+                ts['q'] = self.timeseries_reactive
+            else:
+                ts['q'] = ts['p'] * self.q_sign * tan(acos(self.power_factor))
+            return ts
         else:
             return self._timeseries
 
+    @property
+    def timeseries_reactive(self):
+        """
+        Load reactive power time series used to obtain a reactive power time series
+        instead of calculating it from the assumptions
+
+        Returns
+        -------
+        :pandas:`pandas.Series<series>`
+            series containing reactive power in kvar.
+        """
+        if self._timeseries_reactive is None:
+            pass
+            # logger.debug("No time series " +
+            #                "given. Calculating a reactive power" +
+            #                " time series based on the assumed power factor" +
+            #                ",reactive power mode and the acitve power time" +
+            #                " series.")
+        return self._timeseries_reactive
+
+    @timeseries_reactive.setter
+    def timeseries_reactive(self, timeseries_reactive):
+        """
+        Setting the reactive power time series in kvar.
+
+        Attributes
+        ----------
+        timeseries_reactive : :pandas:`pandas.DataFrame<dataframe>`
```

Comment:
Series or DataFrame?

Reply:
At what point does the generator get its reactive power time series from the TimeSeries object?